### PR TITLE
Migrating Kafka Connect s3 sink by Confluent

### LINKED ISSUE #561: Migrate Kafka Connect S3 sink by Confluent
# What should we add to our content collections?

(more information is useful here! Were you trying to do something particular?)

Migrate the help article 
https://help.aiven.io/en/articles/2413736-aiven-for-apache-kafka-s3-sink-connector-by-confluent

# Is this something you would work on yourself?

[X] I plan to open a pull request for this

### PR DESCRIPTION
**Pull request by ftisiot**
# What changed, and why it matters
 
Migrated Kafka Connect s3 sink by Confluent (https://help.aiven.io/en/articles/2413736-aiven-for-apache-kafka-s3-sink-connector-by-confluent)

Split across multiple pages and fixed previous references to the ones "by Aiven"

Fix #561

### CODE REVIEW

File: docs/products/kafka/kafka-connect/howto/s3-sink-connector-confluent.rst [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        "s3.region": "<AWS_S3_REGION>",
+        "s3.credentials.provider.class": "io.aiven.kafka.connect.util.AivenAWSCredentialsProvider",
+        "storage.class": "io.confluent.connect.s3.storage.S3Storage",
+        "s3.credentials.provider.secret_access_key": "<AWS_USER_SECRET_ACCESS_KEY>",
+        "s3.credentials.provider.access_key_id": "<AWS_USER_ACCESS_KEY_ID>"
+    }
+
+The configuration file contains the following entries:
+
+* ``name``: The connector name
```
Comment (anelook) [inline docs/products/kafka/kafka-connect/howto/s3-sink-connector-confluent.rst:52]: Any reason why we start with capital letter after ":" I think it should be  ": the connector name" Applicable for the whole list

Comment (anelook) [reply docs/products/kafka/kafka-connect/howto/s3-sink-connector-confluent.rst:52]: Also, on a minor side - some sentences have "." at the end, others - not. would be good to be consistent

File: docs/products/kafka/kafka-connect/howto/s3-sink-connector-confluent.rst [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+* ``topics``: The list of Apache Kafka® topics to sink to the S3 bucket
+* ``key.converter`` and ``value.converter``: Data converters, depending on the topic data format. Check the `related documentation <https://docs.confluent.io/5.0.0/connect/kafka-connect-s3/index.html>`_ for more information
+* ``format.class``: Defines the output data format in the S3 bucket. The ``io.confluent.connect.s3.format.bytearray.ByteArrayFormat`` writes messages in binary format.
+* ``flush.size``: Defines how many messages to write per file in the S3 bucket. E.g. setting ``flush.size`` to ``3`` generates a file every three messages in a topic and partition.
+* ``s3.bucket.name``: The name of the S3 bucket
+* ``s3.region``: The AWS region where the S3 bucket has been created
+* ``s3.credentials.provider.class``: The name of the class implementing the ``com.amazonaws.auth.AWSCredentialsProvider`` and ``org.apache.kafka.common.Configurable`` interfaces. Use ``io.aiven.kafka.connect.util.AivenAWSCredentialsProvider``.
+* ``s3.credentials.provider.secret_access_key``: The AWS user access key ID
```
Comment (anelook) [inline docs/products/kafka/kafka-connect/howto/s3-sink-connector-confluent.rst:60]:
```suggestion
* ``s3.credentials.provider.secret_access_key``: The AWS user secret access key
```

File: docs/products/kafka/kafka-connect/howto/s3-sink-connector-confluent.rst [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+* ``s3.credentials.provider.access_key_id``: The AWS user secret access key
```
Comment (anelook) [inline docs/products/kafka/kafka-connect/howto/s3-sink-connector-confluent.rst:61]:
```suggestion
* ``s3.credentials.provider.access_key_id``: The AWS user access key ID
```

File: docs/products/kafka/kafka-connect/reference/s3-sink-additional-parameters-confluent.rst [hunk 4]
```diff
@@ -0,0 +1,57 @@
+S3 sink connector by Confluent naming and data formats
+======================================================
+
+The Apache Kafka Connect® S3 sink connector enables you to move data from an Aiven for Apache Kafka® cluster to Amazon S3 for long term storage. The following document describes advanced parameters defining the naming and data formats.
+
+.. Warning::
+
+    Aiven provides, as managed service, two version of S3 sink connector: one developed by Aiven, another developed by Confluent.
```
Comment (anelook) [inline docs/products/kafka/kafka-connect/reference/s3-sink-additional-parameters-confluent.rst:8]: I'd rephrase, or even remove "As managed service"
```suggestion
    As managed service Aiven provides two version of S3 sink connector: one developed by Aiven, another developed by Confluent. 
```

File: docs/products/kafka/kafka-connect/reference/s3-sink-additional-parameters-confluent.rst [hunk 5]
```diff
@@ -0,0 +1,57 @@
+S3 sink connector by Confluent naming and data formats
+======================================================
+
+The Apache Kafka Connect® S3 sink connector enables you to move data from an Aiven for Apache Kafka® cluster to Amazon S3 for long term storage. The following document describes advanced parameters defining the naming and data formats.
+
+.. Warning::
+
+    Aiven provides, as managed service, two version of S3 sink connector: one developed by Aiven, another developed by Confluent. 
+    
+    This article is about the **Confluent** version. Documentation for the Aiven version is available at the :doc:`dedicated page <s3-sink-additional-parameters>`.
```
Comment (anelook) [inline docs/products/kafka/kafka-connect/reference/s3-sink-additional-parameters-confluent.rst:10]: probably "on the dedicated page"... or "in" :D not sure! 
@TibsAtWork , what is the correct version - "at/on/in the dedicated page"?

File: docs/products/kafka/kafka-connect/reference/s3-sink-additional-parameters.rst [hunk 6]
```diff
@@ -1,7 +1,13 @@
...
-=========================================
+S3 sink connector by Aiven naming and data formats
+==================================================
 
-The Apache Kafka Connect® S3 sink connector enables you to move data from an Aiven for Apache Kafka® cluster to Amazon S3 for long term storage. The following document describes advanced parameters defining the naming and data formats.
+The Apache Kafka Connect® S3 sink connector by Aiven enables you to move data from an Aiven for Apache Kafka® cluster to Amazon S3 for long term storage. The following document describes advanced parameters defining the naming and data formats.
+
+.. Warning::
+
+    Aiven provides, as managed service, two version of S3 sink connector: one developed by Aiven, another developed by Confluent.
```
Comment (anelook) [inline docs/products/kafka/kafka-connect/reference/s3-sink-additional-parameters.rst:8]: Same as above, position of "as managed service" sounds a bit off to me, either move it to the front, or remove (seems not so much relevant to me)

File: docs/products/kafka/kafka-connect/reference/s3-sink-formats.rst [hunk 7]
```diff
@@ -0,0 +1,9 @@
+AWS S3 sink connector naming and data formats
+=============================================
+
+The Apache Kafka Connect® S3 sink connector enables you to move data from an Aiven for Apache Kafka® cluster to Amazon S3 for long term storage. 
+
+Aiven provides, as managed service, two version of the S3 sink connector, each one having different naming and data formats:
```
Comment (anelook) [inline docs/products/kafka/kafka-connect/reference/s3-sink-formats.rst:6]: Same as above, position of "as managed service" sounds a bit off to me, either move it to the front, or remove (seems not so much relevant to me)

File: docs/products/kafka/kafka-connect/howto/s3-sink-connector-confluent.rst [hunk 8]
```diff
@@ -0,0 +1,121 @@
+Create an S3 sink connector by Confluent
+========================================
+
+The Apache Kafka Connect® S3 sink connector by Aiven enables you to move data from an Aiven for Apache Kafka® cluster to Amazon S3 for long term storage.
```
Comment (anelook) [inline docs/products/kafka/kafka-connect/howto/s3-sink-connector-confluent.rst:4]: a minor point - I'd make the names in bold, so that it's a bit easier to read (although, the sentence is short and nice, so not a must). If you decide to change - there are several other places where we use the same.
```suggestion
The **Apache Kafka Connect® S3 sink connector by Aiven** enables you to move data from an **Aiven for Apache Kafka®** cluster to **Amazon S3** for long term storage.
```

**Review by anelook — CHANGES_REQUESTED** [review]
I have a déjà vu, that I read this in the past :D 

I wrote to IT team to get advice on setting up AWS account. So I didn't run it.

Reads very easy to me. Added some minor suggestions and fixes.

File: docs/products/kafka/kafka-connect/reference/s3-sink-additional-parameters-confluent.rst (hunk 5 above)
Comment (ftisiot) [reply docs/products/kafka/kafka-connect/reference/s3-sink-additional-parameters-confluent.rst:10]: I put `in`

File: docs/products/kafka/kafka-connect/reference/s3-sink-formats.rst (hunk 7 above)
Comment (ftisiot) [reply docs/products/kafka/kafka-connect/reference/s3-sink-formats.rst:6]: fixed

File: docs/products/kafka/kafka-connect/reference/s3-sink-formats.rst [hunk 9]
```diff
@@ -0,0 +1,9 @@
+AWS S3 sink connector naming and data formats
+=============================================
+
+The Apache Kafka Connect® S3 sink connector enables you to move data from an Aiven for Apache Kafka® cluster to Amazon S3 for long term storage.
```
Comment (anelook) [inline docs/products/kafka/kafka-connect/reference/s3-sink-formats.rst:4]:
```suggestion
The **Apache Kafka Connect® S3 sink connector** enables you to move data from an **Aiven for Apache Kafka®** cluster to **Amazon S3** for long term storage. 
```

File: docs/products/kafka/kafka-connect/reference/s3-sink-additional-parameters.rst [hunk 10]
```diff
@@ -1,7 +1,13 @@
-S3 sink connector naming and data formats
-=========================================
+S3 sink connector by Aiven naming and data formats
+==================================================
 
-The Apache Kafka Connect® S3 sink connector enables you to move data from an Aiven for Apache Kafka® cluster to Amazon S3 for long term storage. The following document describes advanced parameters defining the naming and data formats.
+The Apache Kafka Connect® S3 sink connector by Aiven enables you to move data from an Aiven for Apache Kafka® cluster to Amazon S3 for long term storage. The following document describes advanced parameters defining the naming and data formats.
```
Comment (anelook) [inline docs/products/kafka/kafka-connect/reference/s3-sink-additional-parameters.rst:4]:
```suggestion
The **Apache Kafka Connect® S3 sink connector** by Aiven enables you to move data from an **Aiven for Apache Kafka®** cluster to **Amazon S3** for long term storage. The following document describes advanced parameters defining the naming and data formats.
```

File: docs/products/kafka/kafka-connect/reference/s3-sink-additional-parameters-confluent.rst [hunk 11]
```diff
@@ -0,0 +1,57 @@
+S3 sink connector by Confluent naming and data formats
+======================================================
+
+The Apache Kafka Connect® S3 sink connector enables you to move data from an Aiven for Apache Kafka® cluster to Amazon S3 for long term storage. The following document describes advanced parameters defining the naming and data formats.
```
Comment (anelook) [inline docs/products/kafka/kafka-connect/reference/s3-sink-additional-parameters-confluent.rst:4]:
```suggestion
The **Apache Kafka Connect® S3 sink connector** enables you to move data from an **Aiven for Apache Kafka®** cluster to **Amazon S3** for long term storage. The following document describes advanced parameters defining the naming and data formats.
```

**Review by anelook — CHANGES_REQUESTED** [review]

**Review by anelook — APPROVED** [review]
